eli5/formatters/html.py

```diff
@@ -0,0 +1,158 @@
+    doc = weighted_spans_data['document']
+    weighted_spans = weighted_spans_data['weighted_spans']
+    char_weights = np.zeros(len(doc))
+    for _, spans, weight in weighted_spans:
+        for start, end in spans:
+            char_weights[start:end] += weight
+    # TODO - can be much smarter, join spans at least
+    # TODO - for longer documents, remove text without active features
+    not_found_weights = sorted(
+        (feature, weight)
+        for feature, weight in weighted_spans_data['not_found'].items()
+        if not np.isclose(weight, 0.))
+    weight_range = max(abs(x) for x in char_weights)
+    if not_found_weights:
+        weight_range = max(weight_range,
+                           max(abs(w) for _, w in not_found_weights))
+    hl_doc = []
+    if not_found_weights:
+        hl_doc.append(' '.join(_colorize(token, weight, weight_range)
+                               for token, weight in not_found_weights))
+    hl_doc.append(''.join(_colorize(token, weight, weight_range)
+                          for token, weight in zip(doc, char_weights)))
+    return ' '.join(hl_doc)
+
+
+def _colorize(token, weight, weight_range):
+    """ Return token wrapped in a span with some styles
+    (calculated from weight and weight_range) applied.
+    """
+    token = html_escape(token)
+    if np.isclose(weight, 0.):
+        return (
+            '<span '
+            'style="opacity: {opacity}"'
+            '>{token}</span>'.format(
+                opacity=_weight_opacity(weight, weight_range),
+                token=token)
+        )
+    else:
+        return (
+            '<span '
+            'style="background-color: {color}; opacity: {opacity}" '
+            'title="{weight:.3f}"'
+            '>{token}</span>'.format(
+                color=_weight_color(weight, weight_range),
+                opacity=_weight_opacity(weight, weight_range),
+                weight=weight,
+                token=token)
+        )
+
+
+def _weight_opacity(weight, weight_range):
+    """ Return opacity value for given weight as a string.
+    """
+    min_opacity = 0.8
+    rel_weight = abs(weight) / weight_range
+    return '{:.2f}'.format(min_opacity + (1 - min_opacity) * rel_weight)
+
+
+def _weight_color(weight, weight_range):
+    """ Return css color for given weight, where the max absolute weight
+    is given by weight_range.
+    """
+    hue = 120 if weight > 0 else 0
+    saturation = 1
+    min_lightness = 0.6
+    lightness = 1.0 - (1 - min_lightness) * abs(weight) / weight_range
+    return 'hsl({}, {:.2%}, {:.2%})'.format(hue, saturation, lightness)
+
+
+def _weight_range(weights):
+    """ Max absolute feature for pos and neg weights.
+    """
+    return max([abs(coef) for key in ['pos', 'neg']
+                for _, coef in weights.get(key, [])] or [0])
+
+
+def _format_unhashed_feature(feature):
+    """ Format unhashed feature: show first (most probable) candidate,
+    display other candidates in title attribute.
```

Comment:
Hm, what do you think about adding it more visible somehow, e.g. adding some symbol to indicate there are known collisions for the feature?